src/child/actions/favourites.js

```diff
@@ -1,19 +1,58 @@
 import { FAVOURITES as ACTION_TYPES } from '../../shared/constants/actionTypes';
+import { selectStock, unselectStock } from './selection';
 import createActionCreator from '../utils/createActionCreator';
+import currentWindowService from '../services/currentWindowService';
 
 export const insertFavouriteAt = createActionCreator((index, code) => ({
     type: ACTION_TYPES.INSERT_FAVOURITE_AT,
     index,
     code
 }));
 
-export const toggleFavourite = createActionCreator((code) => ({
-    type: ACTION_TYPES.TOGGLE_FAVOURITE,
-    code
-}));
-
 export const quandlResponse = createActionCreator((code, name) => ({
     type: ACTION_TYPES.QUANDL_RESPONSE,
     code,
     name
 }));
+
+function toggleFavouriteInWindow(code, windowName) {
+    return {
+        type: ACTION_TYPES.TOGGLE_FAVOURITE,
+        windowName,
+        code
+    };
+}
+
+export function toggleFavourite(code, windowName) {
+    return (dispatch, getState) => {
+        const windowToToggleFavouriteIn = windowName || currentWindowService.getCurrentWindowName();
+        const { favourites, selection } = getState().childWindows[windowToToggleFavouriteIn];
+
+        dispatch(toggleFavouriteInWindow(code, windowToToggleFavouriteIn));
+
+        const isFavourite = favourites.codes.some(favourite => favourite === code);
+        if (selection.code === code && isFavourite) {
+            if (favourites.codes.length) {
```

Comment:
I think this should be `favourites.codes.length > 1` because if there is only 1 favourite and it is unfavourited, then there will not be any favourites left to select. So there need to be at least 2 favourites to make a new selection (1 to unfavourite and the 1 to select as the new selection).